### PR TITLE
refactor overlay

### PR DESCRIPTION
**Pull request by ged-odoo**

### CODE REVIEW

File: addons/html_editor/static/src/core/overlay.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        target: { validate: (el) => el.nodeType === Node.ELEMENT_NODE, optional: true },
+    };
+
+    setup() {
+        this.ref = useRef("root");
+        const overlay = this.props.overlay;
+        this.config = overlay.config;
+        this.offsetY = this.config.offsetY === undefined ? 6 : this.config.offsetY;
+        onMounted(() => {
+            this.ref.el.style.position = "absolute";
```
Comment (Goaman) [inline addons/html_editor/static/src/core/overlay.js]: `<div t-ref="root" class="overlay" style="position: absolute;">` ?

File: addons/html_editor/static/src/core/overlay.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { Component, xml, useRef, onMounted, onWillUnmount } from "@odoo/owl";
+
+export class EditorOverlay extends Component {
+    static template = xml`
+        <div t-ref="root" class="overlay">
+            <t t-component="props.Component" t-props="props.props"/>
+        </div>`;
+
+    static props = {
+        overlay: Object,
```
Comment (FrancoisGe) [inline addons/html_editor/static/src/core/overlay.js:10]: Why not split ?

File: addons/html_editor/static/src/main/powerbox/powerbox_plugin.js [hunk 3]
```diff
@@ -88,14 +82,15 @@ export class PowerboxPlugin extends Plugin {
         this.commandGroups.splice(0, this.commandGroups.length, ...commandGroups);
         this.onApplyCommand = onApplyCommand;
         this.onClose = onClose;
-        this.overlay.open();
+        // this.overlay.open();
```
Comment (Goaman) [inline addons/html_editor/static/src/main/powerbox/powerbox_plugin.js]: 👁️

File: addons/html_editor/static/src/main/table/table_picker.js [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-            this.overlay.close();
+        const editable = this.props.editable;
+        useExternalListener(editable.ownerDocument, "mousedown", (ev) => {
+            this.props.overlay.close();
         });
-        useExternalListener(this.props.el, "keydown", (ev) => {
+        useExternalListener(this.props.editable, "keydown", (ev) => {
             const key = ev.key;
             switch (key) {
                 case "Escape":
```
Comment (ged-odoo) [inline addons/html_editor/static/src/main/table/table_picker.js:25]: oups, close is not a prop

File: addons/html_editor/static/src/main/table/table_ui_plugin.js [hunk 5]
```diff
@@ -28,21 +28,23 @@ export class TableUIPlugin extends Plugin {
...
+            width: "auto",
         });
         /** @type {import("@html_editor/core/overlay_plugin").Overlay} */
         this.rowMenu = this.shared.createOverlay(TableMenu, {
-            type: "row",
-            dispatch: this.dispatch,
+            position: "left",
+            height: "auto",
+            // type: "row",
+            // dispatch: this.dispatch,
```
Comment (Goaman) [inline addons/html_editor/static/src/main/table/table_ui_plugin.js]: 👁️

**Pull request merged**